internal/heartbeat/heartbeat.go

```diff
@@ -262,24 +277,29 @@ func (s *Heartbeat) pushInformation() error {
 		Content:   content,
 		Directive: "heartbeat",
 	}
-
+	log.Debugf("pushInformation: sending content %+v; DeviceID: %s;", content, deviceId)
 	err = s.send(data)
+	log.Debugf("pushInformation: sending content results %s; DeviceID: %s;", err, deviceId)
 
+	if err != nil {
+		s.pushInfoLock.RLock()
+		defer s.pushInfoLock.RUnlock()
+		if s.firstHearbeat {
+			s.data.SetPreviousHardwareInfo(nil)
```

Comment:
isn't it already nil during the first heartbeat?

Reply:
No, the variable is set in `s.data.RetrieveInfo()` and the send is performed after; so if the send fails, we have to be reset the variable value to `nil` to ensure the next call will gather all the information and not just the delta depending on the `Scope` that is set.